src/lightcurve.jl

```diff
@@ -0,0 +1,309 @@
+
+"""
+Abstract type for all light curve implementations.
+"""
+abstract type AbstractLightCurve{T} end
+
+"""
+    EventProperty{T}
+
+A structure to hold additional event properties beyond time and energy.
+"""
+struct EventProperty{T}
+    name::Symbol
+    values::Vector{T}
+    unit::String
+end
+
+"""
+    LightCurveMetadata
+
+A structure containing metadata for light curves.
+"""
+struct LightCurveMetadata
+    telescope::String
+    instrument::String
+    object::String
+    mjdref::Float64
+    time_range::Tuple{Float64,Float64}
+    bin_size::Float64
+    headers::Vector{Dict{String,Any}}
+    extra::Dict{String,Any}
+end
+
+"""
+    LightCurve{T} <: AbstractLightCurve{T}
+
+A structure representing a binned time series with additional properties.
+"""
+struct LightCurve{T} <: AbstractLightCurve{T}
+    timebins::Vector{T}
+    bin_edges::Vector{T}
+    counts::Vector{Int}
+    count_error::Vector{T}
+    exposure::Vector{T}
+    properties::Vector{EventProperty}
+    metadata::LightCurveMetadata
+    err_method::Symbol
+end
+
+"""
+    calculate_errors(counts::Vector{Int}, method::Symbol, exposure::Vector{T}) where T
+
+Calculate statistical uncertainties for count data.
+"""
+function calculate_errors(
+    counts::Vector{Int},
+    method::Symbol,
+    exposure::Vector{T},
+) where {T}
+    if method === :poisson
+        return convert.(T, sqrt.(counts))
+    elseif method === :gaussian
+        return convert.(T, sqrt.(counts .+ 1))
```

Comment:
Gaussian errors need to be given by the user, estimating them like this would be wrong and, in any case, it seems to assume they are Poisson (the sqrt only makes sense in that case, and summing 1 is a quick way to give an error bar when the count rate is 0).